<launch>
  <!-- Arguments -->
  <arg name="bag_name" default="run2_3hz"/>
  <arg name="verbosity" default="3"/>
  
  <arg name="waypoints" default="$(find panoptic_mapping)/cfg/flat_dataset/waypoints.yaml"/>
  <arg name="airsim_cfg" default="$(find panoptic_mapping)/cfg/flat_dataset/airsim.yaml"/>


  <!-- Run -->
  <param name="/use_sim_time" value="true" />
  <!-- Play data -->
  <node pkg="rosbag" type="play" name="rosbag" args="$(find panoptic_mapping)/data/flat_dataset/$(arg bag_name).bag --clock" output="screen"/>


  <!-- RVIZ Visualization -->
  <node type="rviz" name="rviz_voxblox" pkg="rviz" args="-d $(find panoptic_mapping)/cfg/devel_vis.rviz" output="screen"/>
  
  <!-- transforms -->
<!--  <node pkg="tf" type="static_transform_publisher" name="static_imu_tf" args="0 0 0 0 0 0 1 /airsim_drone/Depth_cam /imu 100"/>-->
  
  <!-- Mapper -->
  <node name="panoptic_mapper" pkg="panoptic_mapping" type="panoptic_mapper_node" output="screen" args="-logtostderr --v=$(arg verbosity)">
    <rosparam file="$(find panoptic_mapping)/cfg/devel_mapper_cfg.yaml" />
    <remap from="pointcloud_in" to="/airsim_drone/Pointcloud"/>
  </node>  

</launch>
